Our goal here is to figure out how to use a the higher dimensional embedding model from googles's `Palm API`. It has in the past raised errors difficult to fix.

In [2]:
import os
import google.generativeai as palm

In [3]:
palm.configure(api_key=os.getenv("PALM_API_KEY"))

In [21]:
for model in palm.list_models():
  if 'embedText' in model.supported_generation_methods:
    print(model.name)

models/embedding-gecko-001


In [15]:
embedding = palm.generate_embeddings(model='models/embedding-gecko-001', text="what is an embedding")

In [20]:
len(embedding['embedding'])

768

## Conclusion
There is only one model available at the time. There used to be two but the lower dimensional one (768) is the only one that remains.